packages/atlas-service/src/main.ts

```diff
@@ -604,6 +611,70 @@ export class AtlasService {
     return res.json() as Promise<IntrospectInfo>;
   }
 
+  static async getAggregationFromUserInput({
+    signal,
+    userInput,
+    collectionName,
+    databaseName,
+    schema,
+    sampleDocuments,
+  }: {
+    userInput: string;
+    collectionName: string;
+    databaseName: string;
+    schema?: SimplifiedSchema;
+    sampleDocuments?: Document[];
+    signal?: AbortSignal;
+  }) {
+    throwIfAborted(signal);
+
+    let msgBody = JSON.stringify({
+      userInput,
+      collectionName,
+      databaseName,
+      schema,
+      sampleDocuments,
+    });
+    if (msgBody.length > AI_MAX_REQUEST_SIZE) {
+      // When the message body is over the max size, we try
+      // to see if with fewer sample documents we can still perform the request.
+      // If that fails we throw an error indicating this collection's
+      // documents are too large to send to the ai.
+      msgBody = JSON.stringify({
+        userInput,
+        collectionName,
+        databaseName,
+        schema,
+        sampleDocuments: sampleDocuments?.slice(0, AI_MIN_SAMPLE_DOCUMENTS),
+      });
+      // Why this is not happening on the backend?
```

Comment:
seems totally fine to keep that in the backend

Reply:
The comment got removed, but I am still not sure why this is a client validation and not a backend one

Reply:
We do have this validation on the backend.